keras_nlp/layers/sine_positional_embedding.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sinusoidal Position embedding implementation based on
+`keras.layers.Layer`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class SinePositionEncoding(keras.layers.Layer):
+    """Creates a layer which ouputs a sinusoidal positional embedding for
+    inputs sequences
+
+    This layer calculates the position encoding as a mix of sine and cosine
+    functions with geometrically increasing wavelengths. Defined and formulized
+    in "Attention is All You Need", section 3.5.
+    (https://arxiv.org/abs/1706.03762).
+
+    Args:
+      min_frequency: The minimum frequency for each position embedding,
+                     initalized to 1.0e-4.
+
+    Example:
+    ```python
+    positional_embedding = keras_nlp.layers.SinePositionEncoding()
```

Comment:
Update this to show an example combining a `keras.layers.Embedding` with a `keras_nlp.layers.SinePositionEncoding`, that will be the common usage.